### PR TITLE
Gna cnn2d with pooling support

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - cnn2d with pooling support
 - Add tests

### Tickets:
 - it finalizes tickets:
 - https://jira.devtools.intel.com/browse/CVS-42267
 - https://jira.devtools.intel.com/browse/CVS-51062

### CODE REVIEW

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp [hunk 1]
```diff
@@ -483,6 +483,13 @@ class DataQuantizer<Desc, InferenceEngine::CNNLayer *> : public DataQuantizerBas
                     outData->setPrecision(Desc::mandatory().getInputPrecision());
                 }
             }
+            // for pooling layer output precission is the same as input precission
+            if (LayerInfo(*cnnLayer).isMaxPooling()) {
+                const auto inputPrecission = cnnLayer->insData.front().lock()->getPrecision();
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp:488]: Typo in inputPrecisssion -> should be inputPrecision. Also comments above.

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.hpp [hunk 2]
```diff
@@ -389,20 +381,16 @@ class AMIntelDNN {
...
-                                            uint32_t num_bytes_per_output,
-                                            uint32_t num_pool_size,
-                                            uint32_t num_pool_step,
-                                            uint32_t num_pool_stride,
-                                            bool do_sum_not_max,
-                                            float output_scale_factor,
-                                            void *&ptr_inputs,
-                                            void *&ptr_outputs,
-                                            bool postInitMem);
+        std::array<uint32_t, 3> inCHW,
```
Comment (sirzabek) [inline inference-engine/src/gna_plugin/backend/am_intel_dnn.hpp:384]: Why not keep the indentation aligned with other code?

File: inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp (hunk 1 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/frontend/layer_quantizer.hpp]: fixed

File: inference-engine/src/gna_plugin/backend/am_intel_dnn.hpp (hunk 2 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/backend/am_intel_dnn.hpp]: restored

**Review by sirzabek — APPROVED** [review]

**Pull request merged**